test/helpers/runDeployment.js

```diff
@@ -43,63 +45,71 @@ module.exports = async (account, deploy, deployed, execute, getConfig, keccak256
-    await execute(contractRegistry.registerAddress(asciiToHex('BancorNetwork'), bancorNetwork.address));
-    await execute(contractRegistry.registerAddress(asciiToHex('NetworkSettings'), networkSettings.address));
-
-    await execute(contractRegistry.registerAddress(asciiToHex('ConversionPathFinder'), conversionPathFinder.address));
-    await execute(contractRegistry.registerAddress(asciiToHex('BancorConverterUpgrader'), converterUpgrader.address));
-    await execute(contractRegistry.registerAddress(asciiToHex('BancorConverterRegistry'), converterRegistry.address));
+    await execute(contractRegistry.registerAddress(formatBytes32String('ContractRegistry'), contractRegistry.address));
+    await execute(contractRegistry.registerAddress(formatBytes32String('ConverterFactory'), converterFactory.address));
+    await execute(contractRegistry.registerAddress(formatBytes32String('BancorNetwork'), bancorNetwork.address));
+    await execute(contractRegistry.registerAddress(formatBytes32String('NetworkSettings'), networkSettings.address));
+
+    await execute(
+        contractRegistry.registerAddress(formatBytes32String('ConversionPathFinder'), conversionPathFinder.address)
+    );
+    await execute(
+        contractRegistry.registerAddress(formatBytes32String('BancorConverterUpgrader'), converterUpgrader.address)
+    );
+    await execute(
+        contractRegistry.registerAddress(formatBytes32String('BancorConverterRegistry'), converterRegistry.address)
+    );
     await execute(
-        contractRegistry.registerAddress(asciiToHex('BancorConverterRegistryData'), converterRegistryData.address)
+        contractRegistry.registerAddress(
+            formatBytes32String('BancorConverterRegistryData'),
+            converterRegistryData.address
+        )
     );
 
-    // initialize converter factory
     await execute(converterFactory.registerTypedConverterFactory(standardPoolConverterFactory.address));
 
-    for (const reserve of getConfig().reserves) {
+    for (const reserve of config.reserves) {
         if (reserve.address) {
             const token = await deployed('ERC20', reserve.address);
             const symbol = await token.symbol();
             const decimals = await token.decimals();
             reserves[symbol] = { address: token.address, decimals: decimals };
         } else {
-            const name = reserve.symbol + ' DS Token';
-            const symbol = reserve.symbol;
-            const decimals = reserve.decimals;
-            const supply = decimalToInteger(reserve.supply, decimals);
-            const nonce = await getTransactionCount(account.address);
+            const { symbol, decimals } = reserve;
+            const name = symbol + ' DS Token';
+            const supply = toWei(reserve.supply, decimals);
             const token = await deploy('dsToken-' + symbol, 'DSToken', name, symbol, decimals);
-            if (nonce !== (await getTransactionCount(account.address))) {
```

Comment:
This is of course meaningless when executed via the HH test.

But when executed via the deployment script, this check determines whether or not the previous transaction (deployment of a DS token) had executed "for real".

In other words, it is possible that the script was executed a first time, completed that transaction but failed at a later point, then executed again, in which case that DS token is not redeployed, the nonce doesn't increment, and the script knows that it should not re-issue tokens to the caller (i.e., execute the stuff inside the `if` clause).

Reply:
This shouldn't have been handled via nonce heuristics, but rather via execution steps. In any case, we're going to adapt the whole deployment process to a migratory/incremental process, so this will change as well, so I'd suggest keeping the simplified version.